Community/AssetManagement/lumeta_workflow_page.py

```diff
@@ -0,0 +1,311 @@
+from main_app import app
+
+# application config
+
+# Define global variable to hold handle to API object
+api = None
+
+#
+# GET, PUT or POST
+#
+@route(app, '/lumeta/getnetworklist', methods=['GET', 'PUT', 'POST'])
+@util.rest_workflow_permission_required('lumeta_workflow_page')
+@util.rest_exception_catcher
+def get_networks_get_networks_page():
+    # are we authenticated?
+    g.user.logger.info('SUCCESS')
+    configurations = None
+    configurations_json = []
+    if g.user:
+        configurations = g.user.get_api().get_configurations()
+        for c in configurations:
+            print (c)
+            configuration_json = {"id": c.get_id(), "name": c.get_name()}
+            configurations_json.append(configuration_json)
+        return jsonify(configurations_json)
+
+
+@route(app, '/lumeta/getiplist', methods=['GET', 'PUT', 'POST'])
+@util.rest_workflow_permission_required('lumeta_workflow_page')
+@util.rest_exception_catcher
+def getiplist_getiplist_page():
+    # are we authenticated?
+    g.user.logger.info('SUCCESS')
+    networks = []
+
+    # Return object that contains all the networks (and eventually all ip addresses)
+    # list of all properties objects
+    ip_addresses = []
+    # If name is given, use get_configuration(name)
+    if g.user:
+        configurations = g.user.get_api().get_configurations()
+        for c in configurations:
+            print(c)
+            configuration_json = {"id": c.get_id(), "name": c.get_name()}
+
+            # FIXME - need code to get network list from configuration id. Is there a call to get children_of_types
+            #  (['IP4Block', 'IP4Network', 'IP6Block', 'IP6Network'
+            # use get_by_object_types(*,  ['IP4Block', 'IP4Network', 'IP6Block', 'IP6Network']) - returns flat list
+            # We might want to request IP4Network, IP6Network
+            # FIXME - extract below code in a function and call it for IP4Block and IP6Block
+            try:
+                for nw in c.get_children_of_type('IP4Block'):
+                    print(nw)
+                    # get all blocks and networks for block
+                    for n in g.user.get_api().get_by_object_types(nw.get_property('CIDR'),
+                                                                  ['IP4Network', 'IP4Block', 'IP6Network', 'IP6Block']):
+                        if '6' in n.get_type():
+                            networks.append({'network_id': n.get_id(), 'display_text': n.get_properties()['prefix']})
+                            ip_addresses.extend(calculate_block_stats(n, c.get_id(), c.get_name()))
+                        else:
+                            networks.append({'network_id': n.get_id(), 'display_text': n.get_properties()['CIDR']})
+                            ip_addresses.extend(calculate_block_stats(n, c.get_id(), c.get_name()))
+
+            except Exception as e:
+                app.loggererror('get_subnets: ' + e.message)
+    return jsonify(ip_addresses)
```

Comment:
There doesn't appear to be anything letting the end user know that there's an incomplete or errored dataset